code/debris/debris.cpp

```diff
@@ -1312,7 +1312,11 @@ void create_generic_debris(object* ship_objp, const vec3d* pos, float min_num_de
 	vec3d create_pos = *pos;
 	for (int i = 0; i < num_debris; i++) {
 		int model_num = use_ship_debris ? Ship_info[Ships[ship_objp->instance].ship_info_index].generic_debris_model_num : -1;
-		debris_create(ship_objp, model_num, -1, &create_pos, pos, 0, speed_mult);
+		// only proceed if generic debris is allowed --wookeejedi
+		if ( !(Ship_info[Ships[ship_objp->instance].ship_info_index].flags[Ship::Info_Flags::Disable_all_generic_explosion_debris]) &&
+			((model_num >= 0) || !(Disable_all_noncustom_generic_debris)) ) {
+				debris_create(ship_objp, model_num, -1, &create_pos, pos, false, speed_mult);
+		}
```

Comment:
I think this change should actually be removed.  Consider that it is redundant to the code you added in shiphit.cpp, and it also checks a different flag: explosion debris vs impact debris.  Rather than checking the flag here, the flag should be checked in any place that `create_generic_debris` is likely to be called.

Reply:
Ah good point, I pushed an update which cleans up this logic. Happy to continue editing as warranted, thanks!